src/components/LineSeries.tsx

```diff
@@ -0,0 +1,48 @@
+import { extent } from 'd3-array';
+import { scaleLinear } from 'd3-scale';
+import { line } from 'd3-shape';
+import React, { CSSProperties, useEffect } from 'react';
+
+import { useDispatchContext, usePlotContext } from '../hooks';
+import type { LineSeriesProps, Series } from '../types';
+
+interface LineSeriesRenderProps {
+  width?: number;
+  height?: number;
+  data: Series[];
+  lineStyle?: CSSProperties;
+}
+
+export default function LineSeries(props: LineSeriesProps) {
+  const { data, label, ...otherProps } = props;
```

Comment:
here you can add a `const [id] = useState(generateId)`

(generateId can just be a function that creates an id with an incremented number like `series-${++number}`)